README.md

```diff
@@ -0,0 +1,754 @@
+
+subselector = "." CHAR string                           ; Dotted field selector
+            / *1(".") "[\" DQUOTE string "\" DQUOTE "]" ; Explicit field selector
+            / *1(".") "[" integer "]"                   ; Index selector
+            / *1(".") "[]"                              ; Collection values // FIXME doble check code
+
+;; SPECIAL LITERALS
+
+pattern     = DQUOTE string DQUOTE ; Reminder: IPLD strings are UTF-8
+number      = integer / float
+```
+ 
+## Comparisons
+
+| Operator | Argument(s)                    | Example                          |
+|----------|--------------------------------|----------------------------------|
+| `==`     | `Selector, IPLD`               | `["==", ".a", [1, 2, {"b": 3}]]` |
+| `<`      | `Selector, (integer \| float)` | `["<",  ".a", 1]`                |
+| `<=`     | `Selector, (integer \| float)` | `["<=", ".a", 1]`                |
+| `>`      | `Selector, (integer \| float)` | `[">",  ".a", 1]`                |
+| `>=`     | `Selector, (integer \| float)` | `[">=", ".a", 1]`                |
+
+Literal equality (`==`) MUST match the resolved selecor to entire IPLD argument. This is a "deep comparison".
+
+Numeric inequalities MUST be agnostic to numeric type. In other words, the decimal representation is considered equivalent to an integer (`1 == 1.0 == 1.00`). Attempting to compare a non-numeric type MUST return false and MUST NOT throw an exception.
+
+## Glob Matching
+
+| Operator | Argument(s)         | Example                             |
+|----------|---------------------|-------------------------------------|
+| `match`  | `Selector, Pattern` | `["==", ".email", "*@example.com"]` |
+
+Glob patterns MUST only include one specicial character: `*` ("wildcard"). There is no single character matcher. As many `*`s as desired MAY be used. Non-wildcard `*`-literals MUST be escaped (`"\*"`). Attempting to match on a non-string MUST return false and MUST NOT throw an exception.
+
+The wildcard represents zero-or-more characters. The following string literals MUST pass validation for the pattern `"Alice\*, Bob*, Carol.`:
+
+* `"Alice*, Bob, Carol."`
+* `"Alice*, Bob, Dan, Erin, Carol."`
+* `"Alice*, Bob  , Carol."`
+* `"Alice*, Bob*, Carol."`
+
+The following MUST NOT pass validation for that same pattern:
+
+* `"Alice*, Bob, Carol"` (missing the final `.`)
+* `"Alice*, Bob*, Carol!"` (final `.` MUST NOT be treated as a wildcard)
+* `"Alice, Bob, Carol."` (missing the `*` after `Alice`)
+* `"Alice Cooper, Bob, Carol."` (the `*` after `Alice` is an escaped literal in the pattern)
+* `" Alice*, Bob, Carol. "` (whitespace in the pattern is significant)
+
+## Connectives
+
+Connectives add context to their enclosed statement(s).
+
+| Operator | Argument(s)   | Example                                    |
+|----------|---------------|--------------------------------------------|
+| `not`    | `Statement`   | `["not", [">", ".a", 1]]`                  |
```

Comment:
> be able to enclose multiple statements

That may be confusing, no? Does it add an implicit `and`? Datomic likely does an implicit `or` due to how DNF works in datalog (which this is not)

Reply:
@Gozala

> Restricting negation to single cause is unnecessarily restricting and leaves undefined behavior when multiple clause may be present. 

Also I disagree about undefined behaviour when multiple clauses are present: that case is disallowed since multiple clauses after a `not` matches no forms.